machines/openIdBle/scan.ts

```diff
@@ -753,17 +750,25 @@ export const scanMachine =
       },
 
       services: {
-        checkBluetoothService: () => (callback) => {
-          const subscription = BluetoothStateManager.onStateChange((state) => {
-            if (state === 'PoweredOn') {
+        checkBluetoothPermission: () => async (callback) => {
+          // wait a bit for animation to finish when app becomes active
+          await new Promise((resolve) => setTimeout(resolve, 250));
+
+          try {
+            let permission =
+              Platform.OS === 'ios'
+                ? PERMISSIONS.IOS.BLUETOOTH_PERIPHERAL
+                : PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION;
+            let response = await check(permission);
+            if (response === RESULTS.GRANTED) {
               callback(model.events.BLUETOOTH_ENABLED());
```

Comment:
Can we rename events to ALLOWED and DENIED to reduce confusion between permission and state